packages/komencikit/src/verifyWallet.ts

```diff
@@ -0,0 +1,66 @@
+import { Address, normalizeAddress } from '@celo/base'
+import { Err, Ok, Result } from '@celo/base/lib/result'
+import { ContractKit } from '@celo/contractkit'
+import { GET_IMPLEMENTATION_ABI } from '@celo/contractkit/lib/governance/proxy'
+import Proxy from '@celo/protocol/build/contracts/Proxy.json'
+import {
+  InvalidBytecode,
+  InvalidImplementation,
+  InvalidSigner,
+  WalletValidationError,
+} from './errors'
+
+export const verifyWallet = async (
+  contractKit: ContractKit,
+  walletAddress: Address,
+  allowedImplementations: Address[],
+  expectedSigner: Address
+): Promise<Result<true, WalletValidationError>> => {
+  const code = await contractKit.web3.eth.getCode(walletAddress)
+  // XXX: I'm unsure whether this is safe or if we should store the
+  // bytecode as a constant in `mobile` and pass it into KomenciKit
```

Comment:
Yeah I guess that was more about how to track the implementation's bytecode itself. We were between using bytecode digests or the actual address of the contract. I was inclining towards bytecode digest because I was under the wrong impression that it would be constant across networks and be easier to manage on Valora - it wouldn't need a different mapping per network. But realistically because we're using library linking it's gonna have different bytecodes therefore we're ok with Valora referring to a MTW implementation by the address, and Komenci verifying server side that address maps to a known implementation.

Here I'm actually verifying the Proxy contract bytecode to ensure that the Proxy itself is what we expect. I suspect we won't need to change the Proxy anytime soon so I think it's safe to extract that from the compilation output. But I'll ping the CAP team to figure out if there are holes in this approach.